File: test/rice_cooker.rb
require 'minitest/autorun'
require_relative '../core/rice_cooker'

class Test < Minitest::Test
  def test_add_water_ok()
    cooker = RiceCooker.new

    cooker.add_water

    assert(1 == cooker.water_level)
  end

  def test_decrease_water_level_ok
    cooker = RiceCooker.new

    cooker.add_water
    cooker.decrease_water_level

    assert(0 == cooker.water_level)
  end

  def test_add_rice_ok
    cooker = RiceCooker.new

    cooker.add_rice

    assert(1 == cooker.rice_amount)
  end

  def test_decrease_rice_amount_ok
    cooker = RiceCooker.new

    cooker.add_rice
    cooker.decrease_rice_amount

    assert(0 == cooker.rice_amount)
  end

end

File: test/rice_cooker.dart
import 'package:test/test.dart';
import '../core/rice_cooker.dart';

void main() {
  group('RiceCooker add test', () {
    test('add water ok', () {
      var rice_cooker = RiceCooker();

      rice_cooker.addWater();

      expect(1, rice_cooker.waterLevel);
    });

    test('decrease water amount ok', () {
      var rice_cooker = RiceCooker();
      rice_cooker.addWater();

       rice_cooker.decreaseWaterLevel();

      expect(0, rice_cooker.waterLevel);
    });

    test('add rice ok', () {
      var rice_cooker = RiceCooker();

      rice_cooker.addRice();

      expect(1, rice_cooker.riceAmount);
    });

    test('decrease rice amount ok', () {
     var rice_cooker = RiceCooker();
     rice_cooker.addRice();

     rice_cooker.decreaseRiceAmount();

      expect(0, rice_cooker.riceAmount);
    });
  });
}